arc-api/server/models/subscription.js

```diff
@@ -0,0 +1,272 @@
+var request = require('request');
+var KeyStore = require('./key-store');
+var debug = require('debug')('strong-arc:subscription');
+
+// Extract the access token from the request
+function getAccessToken(req) {
+  var token = req.query.access_token;
+  if (token) {
+    return token;
+  }
+  token = req.get('authorization');
+  if (typeof token === 'string') {
+    // Add support for oAuth 2.0 bearer token
+    // http://tools.ietf.org/html/rfc6750
+    if (token.indexOf('Bearer ') === 0) {
+      token = token.substring(7);
+      // Decode from base64
+      var buf = new Buffer(token, 'base64');
+      token = buf.toString('utf8');
+    } else if (/^Basic /i.test(token)) {
+      token = token.substring(6);
+      token = (new Buffer(token, 'base64')).toString('utf8');
+      // The spec says the string is user:pass, so if we see both parts
+      // we will assume the longer of the two is the token, so we will
+      // extract "a2b2c3" from:
+      //   "a2b2c3"
+      //   "a2b2c3:"   (curl http://a2b2c3@localhost:3000/)
+      //   "token:a2b2c3" (curl http://token:a2b2c3@localhost:3000/)
+      //   ":a2b2c3"
+      var parts = /^([^:]*):(.*)$/.exec(token);
+      if (parts) {
+        token = parts[2].length > parts[1].length ? parts[2] : parts[1];
+      }
+    }
+    return token;
+  }
+}
+
+module.exports = function(Subscription) {
+  var store = new KeyStore();
+  /**
+   * Get all product subscriptions for a given user
+   * @param {Number|String} userId User id
+   * @param {String} mode Operation mode: online/offline/...
+   * @param {Request} req HTTP request object
+   * @param {Function} cb callback
+   */
+  Subscription.getSubscriptions = function(userId, mode, req, cb) {
+    if (mode === 'offline') {
+      // Read from the local key store
+      return store.load(userId, cb);
+    }
+    var url = Subscription.settings.authUrl;
+    request.get({
+      url: url + 'users/' + userId + '/subscriptions',
+      qs: {
+        access_token: getAccessToken(req)
+      },
+      json: true
+    }, function(err, res, body) {
+      if (err) {
+        if (mode === 'online') {
+          // Report error for online mode
+          return cb(err, body);
+        } else {
+          // Fall back to offline mode
+          return store.load(userId, function(err, content) {
+            if (err) {
+              return cb(err);
+            } else {
+              return (content && content.licenses) || [];
+            }
+          });
+        }
+      } else {
+        // Refresh the local key store
+        store.save({licenses: body, modified: new Date()}, userId, function(err) {
+          cb(err, body);
+        });
+      }
+    });
+  };
+
+  function reloadSubscriptions(userId, accessToken, result, cb) {
+    Subscription.getSubscriptions(userId, 'online', {
+      query: {access_token: accessToken}
```

Comment:
Maybe force `accessToken` to be atleast `null`... I think we remove undefined properties and may match more than you want... Although I'm not 100% sure.